network/peer/peer.go

```diff
@@ -626,6 +629,10 @@ func (p *peer) writeMessage(writer io.Writer, msg message.OutboundMessage) {
 	now := p.Clock.Time()
 	p.storeLastSent(now)
 	p.Metrics.Sent(msg)
+
+	if msg.Op() == message.PingOp {
+		atomic.StoreInt64(&p.lastPingSent, now.UnixMilli())
+	}
```

Comment:
Couple questions for this:
1. Is it a problem that we are setting this variable _after_ we have sent the message? Technically seems racy between handling the pong and setting this ping time here.
2. Would it make more sense to set this where we send the ping message (`peer.sendNetworkMessages`) rather than here? That would avoid needing to actually inspect the op type.

Reply:
I thought about (1) but we don't send the ping there, we just enqueue it to be being sent. It would be inaccurate because if we have some number of fat messages enqueued before us, it would skew the statistics. 

I moved the code to right before we write the message down the wire.